homeassistant/components/bittrex/__init__.py

```diff
@@ -0,0 +1,228 @@
+
+    @staticmethod
+    def _prep_tickers(asset_currencies, tickers):
+        """Prepare tickers data."""
+
+        asset_tickers_dict = {}
+
+        for asset in asset_currencies:
+            # Skip the ASSET_VALUE_BASE as we calculate it differently
+            if asset != ASSET_VALUE_BASE:
+                currency = f"{asset}-{ASSET_VALUE_BASE}"
+
+                # Flip currency format as it differs from other currencies
+                if currency == "EUR-USDT":
+                    currency = "USDT-EUR"
+
+                if currency not in asset_tickers_dict:
+                    asset_tickers_dict[currency] = {}
+                    ticker_details = tickers[currency]
+                    asset_tickers_dict[currency].update(ticker_details)
+
+        return asset_tickers_dict
+
+    @staticmethod
+    def _prep_balances(balances, tickers):
+        """Prepare balances data."""
+
+        balances_dict = {}
+
+        for balance in balances:
+            if (
+                balances[balance]["currencySymbol"] not in balances_dict
+                and balances[balance]["currencySymbol"] != "BTXCRD"
+            ):
+                balances_dict[balances[balance]["currencySymbol"]] = {}
+                balances_dict[balances[balance]["currencySymbol"]].update(
+                    balances[balance]
+                )
+                base_asset_ticker_details = None
+
+                total_balance = float(balances[balance]["total"])
+
+                if ASSET_VALUE_BASE not in balances[balance]["currencySymbol"]:
+                    # Prevent that we try to search ASSET_VALUE_BASE+ASSET_VALUE_BASE (e.g. BTCBTC)
+                    base_asset_symbol = (
+                        f"{balances[balance]['currencySymbol']}-{ASSET_VALUE_BASE}"
+                    )
+
+                    # Flip currency format as it differs from other currencies
+                    if base_asset_symbol == "EUR-USDT":
+                        base_asset_symbol = "USDT-EUR"
+
+                    base_asset_ticker_details = tickers[base_asset_symbol]
+
+                else:
+                    balances_dict[balances[balance]["currencySymbol"]][
+                        "asset_value_in_base_asset"
+                    ] = total_balance
+
+                if base_asset_ticker_details:
+                    # If we can find a pair with ASSET_VALUE_BASE, include it in the dict
+                    balances_dict[balances[balance]["currencySymbol"]][
+                        ASSET_VALUE_BASE
+                    ] = {}
+                    balances_dict[balances[balance]["currencySymbol"]][
+                        ASSET_VALUE_BASE
+                    ].update(base_asset_ticker_details)
+                    balances_dict[balances[balance]["currencySymbol"]][
+                        "asset_value_in_base_asset"
+                    ] = total_balance * float(
+                        base_asset_ticker_details["lastTradeRate"]
+                    )
```

Comment:
The if statement at the end looks like immediately related to the first if statement, so you probably don't need it

```suggestion
                total_balance = float(balances[balance]["total"])

                if ASSET_VALUE_BASE not in balances[balance]["currencySymbol"]:
                    # Prevent that we try to search ASSET_VALUE_BASE+ASSET_VALUE_BASE (e.g. BTCBTC)
                    base_asset_symbol = (
                        f"{balances[balance]['currencySymbol']}-{ASSET_VALUE_BASE}"
                    )

                    # Flip currency format as it differs from other currencies
                    if base_asset_symbol == "EUR-USDT":
                        base_asset_symbol = "USDT-EUR"

                    base_asset_ticker_details = tickers[base_asset_symbol]
                    
                    # If we can find a pair with ASSET_VALUE_BASE, include it in the dict
                    balances_dict[balances[balance]["currencySymbol"]][
                        ASSET_VALUE_BASE
                    ] = {}
                    balances_dict[balances[balance]["currencySymbol"]][
                        ASSET_VALUE_BASE
                    ].update(base_asset_ticker_details)
                    balances_dict[balances[balance]["currencySymbol"]][
                        "asset_value_in_base_asset"
                    ] = total_balance * float(
                        base_asset_ticker_details["lastTradeRate"]
                    )

                else:
                    balances_dict[balances[balance]["currencySymbol"]][
                        "asset_value_in_base_asset"
                    ] = total_balance
```